### ISSUE TITLE
Code for Section 4.4. Quantitative Evaluation of Interpretability

### ISSUE THREAD

**Issue by expectopatronum** [zhiCHEN96/ConceptWhitening#10]
Hi again!
I am trying to run some experiments on my own and was looking for the code that was used in Section 4.4 to create Figure 6. Can I find this somewhere in the repo?

Best regards
Verena

**Comment by zhiCHEN96** [comment]
Yes, the code for calculating the concept AUC is the **_plot_auc_cw_** function in **_plot_functions.py_**. The function will calculate the concept purity (test AUC) for the concept axes in CW module when it's applied to different layers in CNN. It'll print and save the mean and std of the concept purity scores. Before running this, you'll need to make sure you have saved multiple CNN weights with CW applied to different layer, because it will load these different model weights and calculate the score for each of them.

You can also compare it with other baselines by running **_plot_auc_lm_** and **_plot_auc_filter_**, i.e. the baselines we compared with in our paper. Then you can run the **_plot_auc_** function to show these results on the same plot, as shown in Figure 6.

Our code calling these functions are line **522-535** in **_train_places.py_**. You may want to just uncomment them, and modify the inputs to run your case.
